athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQuerySqlUtils.java

```diff
@@ -72,224 +52,41 @@ private BigQuerySqlUtils()
-
-                    // Right side padding with required zeros
-                    val = String.format("%-26s", tempVal).replace(' ', '0').replace("T", " ");
-                    return QueryParameterValue.dateTime(val);
-                }
-                else {
-                    // date search: date parameter used in where clause will come as days so it will be converted to date
-                    long days = Long.parseLong(val);
-                    long milliseconds = TimeUnit.DAYS.toMillis(days);
-                    return QueryParameterValue.date(new SimpleDateFormat("yyyy-MM-dd").format(new Date(milliseconds)));
-                }
-            case Time:
-            case Timestamp:
-            case Interval:
-            case Binary:
-            case FixedSizeBinary:
-            case Null:
-            case Struct:
-            case List:
-            case FixedSizeList:
-            case Union:
-            case NONE:
-                throw new UnsupportedOperationException("The Arrow type: " + arrowType.getTypeID().name() + " is currently not supported");
-            default:
-                throw new IllegalArgumentException("Unknown type has been encountered during range processing: " + columnName +
-                        " Field Type: " + arrowType.getTypeID().name());
-        }
+        BigQueryQueryBuilder queryBuilder = queryFactory.createQueryBuilder();
+        
+        String sql = queryBuilder
+                .withTableName(tableName)
+                .withProjection(schema)
+                .withConjuncts(schema, constraints)
+                .withOrderByClause(constraints)
+                .withLimitClause(constraints)
+                .build();
+        
+        // Copy the parameter values from the builder to the provided list
+        parameterValues.clear();
+        parameterValues.addAll(queryBuilder.getParameterValues());
+
+        return sql;
     }
 
     /**
-     * Based on com.amazonaws.athena.connectors.jdbc.manager.JdbcSplitQueryBuilder.extractOrderByClause() method
-     * @param constraints
-     * @return a string representing ORDER BY clause or an empty string if there is no ORDER BY clause
+     * Generic method to render any string template with parameters
+     *
+     * @param templateName The name of the template to render
+     * @param params Parameters in pairs: (key, value, key, value, ...)
+     * @return The rendered template string
      */
-    private static String extractOrderByClause(Constraints constraints)
+    public static String renderTemplate(String templateName, Object... params)
     {
-        List<OrderByField> orderByClause = constraints.getOrderByClause();
-        if (orderByClause == null || orderByClause.size() == 0) {
-            return "";
+        ST template = queryFactory.getQueryTemplate(templateName);
+        
+        // Add parameters in pairs: (key, value, key, value, ...)
```

Comment:
If we expect a specific structure of parameters, (key-pairs in this case), then why can't we just use a datastructure to capture this instead of relying on logic in a for-loop?